cli/how-to-deploy-rest.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+## IMPORTANT: this file and accompanying assets are the source for snippets in https://docs.microsoft.com/azure/machine-learning! 
+## Please reach out to the Azure ML docs & samples team before before editing for the first time.
+
+# <installation>
+# az extension add -n ml
+# </installation>
+
+# <create environment variables>
+SUBSCRIPTION_ID="7ab7d5bc-5d9e-47ef-80e6-2dffa8ca83a1"
+RESOURCE_GROUP="trmccorm-centraluseuap"
+WORKSPACE="trmccorm-centraluseuap"
+API_VERSION="2021-03-01-preview"
+
+TOKEN=$(az account get-access-token | jq -r ".accessToken")
+
+export AZURE_STORAGE_ACCOUNT="trmccormcentra1277620275"
+export AZURE_STORAGE_KEY=$(az storage account keys list --account-name $AZURE_STORAGE_ACCOUNT | jq '.[0].value')
+AZUREML_DEFAULT_CONTAINER="azureml-blobstore-2e2e441d-f57b-41fa-bd88-49136cef6140"
+#</create environment variables>
+
+# <create resource group>
+# az group create -n azureml-examples-cli -l eastus
+# </create resource group>
+
+# <create workspace>
+# az ml workspace create --name main -g azureml-examples-cli
+# </create workspace>
+
+# <configure-defaults>
+az configure --defaults workspace=$WORKSPACE
+az configure --defaults location="centraluseuap"
+az configure --defaults group=$RESOURCE_GROUP
+# </configure-defaults>
+
+# define how to wait
+wait_for_completion () {
+    operationid=$(echo $1 | grep -Fi Azure-AsyncOperation | sed "s/azure-asyncoperation: //" | tr -d '\r')
+    # TODO error handling here
+    operation_status="unknown"
+
+    while [[ $operation_status != "Succeeded" && $operation_status != "Failed" ]]
+    do
+        $echo "Getting operation status from: $operationid"
+        operation_result=$(curl --location --request GET $operationid --header "Authorization: Bearer $TOKEN")
+        # TODO error handling here
+        operation_status=$(echo $operation_result | jq -r ".status")
+        echo "Current operation status: $operation_status"
+        sleep 5
+    done
+}
+
+# delete endpoint
+curl --location --request DELETE "https://management.azure.com/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE_NAME/onlineEndpoints/my-endpoint?api-version=$API_VERSION" \
+--header "Content-Type: application/json" \
+--header "Authorization: Bearer $TOKEN"
+
+# TODO: we can get the default container from listing datastores
+# TODO using the latter two as env vars shouldn't be necessary
+az storage blob upload-batch -d $AZUREML_DEFAULT_CONTAINER/score \
+ -s endpoints/online/model-1/onlinescoring --account-name $AZURE_STORAGE_ACCOUNT --account-key $AZURE_STORAGE_KEY
```

Comment:
According to the docs, `$AZURE_STORAGE_ACCOUNT` and `$AZURE_STORAGE_KEY` shouldn't be necessary if they are environment variables but I've never been able to get this to work on Linux